gitlab/resource_gitlab_deploy_key.go

```diff
@@ -35,7 +39,33 @@ func resourceGitlabDeployKey() *schema.Resource {
 				Required: true,
 				ForceNew: true,
 				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
-					return old == strings.TrimSpace(new)
+					var oldPubKey, newPubKey ssh.PublicKey
+					var err error
+
+					switch new == "" {
+					case true:
+						return false
+					case false:
+						newPubKey, _, _, _, err = ssh.ParseAuthorizedKey([]byte(new))
+						if err != nil {
+							panic(err)
+						}
+					}
+
+					switch old == "" {
+					case true:
+						dummyKey, err := rsa.GenerateKey(rand.Reader, 1024)
+						if err != nil {
+							panic(err)
+						}
+						oldPubKey, err = ssh.NewPublicKey(&dummyKey.PublicKey)
+					case false:
+						oldPubKey, _, _, _, err = ssh.ParseAuthorizedKey([]byte(old))
+						if err != nil {
+							panic(err)
+						}
+					}
+					return string(bytes.TrimSpace(oldPubKey.Marshal())) == string(bytes.TrimSpace(newPubKey.Marshal()))
```

Comment:
don't you need to empty the  comment first?

Reply:
you're right, after inspection, the Marshal() method returns everything.  Let me look into it.

Reply:
Actually, the Marshal() method only returns the public key, so no need for further trimming.